src/memory/memory.cc

```diff
@@ -1,3 +1,45 @@
 #include "memory/memory.hh"
 
-namespace sim {}
+namespace sim {
+
+Memory::Memory() {}
+
+Word Memory::loadWord(AddrType addr) {
+
+  stats.numLoads++;
+  Word word = kDummyWord;
+  try {
+    word = mem.at(addr);
+  } catch (const std::out_of_range &) {
+    word = pageFaultHandle(addr);
+  }
```

Comment:
Exception handling is overkill here. It's better to use `std::unordered_map::contains` or `std::unordered_map::find`.

Reply:
Oh, yes, forgot about this. Reworked it in new commit